src/View/ViewLayerData.php

```diff
@@ -0,0 +1,254 @@
+<?php
+
+namespace SilverStripe\View;
+
+use BadMethodCallException;
+use InvalidArgumentException;
+use IteratorAggregate;
+use SilverStripe\Core\ClassInfo;
+use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\Model\ModelData;
+use SilverStripe\Model\ModelDataCustomised;
+use SilverStripe\ORM\FieldType\DBClassName;
+use Stringable;
+use Traversable;
+
+class ViewLayerData implements IteratorAggregate, Stringable
+{
+    use Injectable;
+
+    public const TYPE_PROPERTY = 'property';
+
+    public const TYPE_METHOD = 'method';
+
+    public const TYPE_ANY = 'any';
```

Comment:
As discussed elsewhere, we don't actually _use_ these right now, but they exist to:
1. Keep a door open for us to use them in future enhancements
2. Allow projects and modules to choose to be more strict about how values are accessed in their view layer